wbooks/src/app/components/SingleBook/index.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Image, Text, View } from 'react-native';
+
+import { styles } from './styles';
+
+interface Props {
+  bookCover: string | null;
```

Comment:
I didn't know this one, thanks! Will absolutely use it from now on.

Reply:
This results in a TS error, if the prop type is `ImageSourcePropType`, the prop should be passed as `bookCover={{uri: BOOKS_MOCK[0].imageUrl}}` (per the RN type definitions), I tried it this way and it worked, but when the url is `null` it won't display the placeholder image because the value of the prop turns out to be the object with the `uri` key with a `null` value (and therefore the condition will never be falsy). This can be avoided typing the `bookCover` prop as a string.